apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -1541,6 +1655,113 @@ impl Telemetry {
     }
 }
 
+#[derive(Debug, Clone)]
+struct CacheAttributes {
+    subgraph_name: Arc<String>,
+    headers: http::HeaderMap,
+    hashed_query: Arc<String>,
+    // Typename + hashed_representation
+    representations: Vec<(Arc<String>, Value)>,
+}
+
+#[derive(Debug, Hash, Clone)]
+struct CacheKey {
+    representation: Value,
+    typename: Arc<String>,
+    query: Arc<String>,
+    subgraph_name: Arc<String>,
+    hashed_headers: Arc<String>,
+}
+
+// Get typename and hashed representation for each representations in the subgraph query
+fn extract_cache_attributes(
+    representations: &[Value],
+) -> Result<Vec<(Arc<String>, Value)>, BoxError> {
+    let mut res = Vec::new();
+    for representation in representations {
+        let opt_type = representation
+            .as_object()
+            .and_then(|o| o.get(TYPENAME))
+            .ok_or("missing __typename in representation")?;
+        let typename = opt_type.as_str().unwrap_or("");
+
+        res.push((Arc::new(typename.to_string()), representation.clone()));
+    }
+    Ok(res)
+}
+
+struct CacheCounter {
+    primary: Bloom<CacheKey>,
+    secondary: Bloom<CacheKey>,
+    created_at: Instant,
+    ttl: Duration,
+}
+
+impl CacheCounter {
+    fn new(ttl: Duration) -> Self {
+        Self {
+            primary: Self::make_filter(),
+            secondary: Self::make_filter(),
+            created_at: Instant::now(),
+            ttl,
+        }
+    }
+
+    fn make_filter() -> Bloom<CacheKey> {
+        // the filter is around 4kB in size (can be calculated with `Bloom::compute_bitmap_size`)
+        Bloom::new_for_fp_rate(10000, 0.2)
+    }
+
+    fn record(
+        &mut self,
+        query: Arc<String>,
+        subgraph_name: Arc<String>,
+        hashed_headers: Arc<String>,
+        representations: Vec<(Arc<String>, Value)>,
+    ) {
+        if self.created_at.elapsed() >= self.ttl {
+            self.clear();
+        }
+
+        // typename -> (nb of cache hits, nb of entities)
+        let mut seen: HashMap<Arc<String>, (usize, usize)> = HashMap::new();
+        for (typename, representation) in representations {
+            let cache_hit = self.check(&CacheKey {
+                representation,
+                typename: typename.clone(),
+                query: query.clone(),
+                subgraph_name: subgraph_name.clone(),
+                hashed_headers: hashed_headers.clone(),
+            });
+
+            let seen_entry = seen.entry(typename.clone()).or_default();
+            if cache_hit {
+                seen_entry.0 += 1;
+            }
+            seen_entry.1 += 1;
+        }
+
+        for (typename, (cache_hit, total_entities)) in seen.into_iter() {
+            ::tracing::info!(
+                histogram.apollo.router.operations.entity = (cache_hit as f64 / total_entities as f64) * 100f64,
```

Comment:
I don't think this is the right key:
`histogram.apollo.router.operations.entity` to me should indicate that an entity was used in the query vs was it cachable or not.

Suggest:
`histogram.apollo.router.operations.entity.cacheable`?

Reply:
`histogram.apollo.router.operations.entity.cache_hit` may be better